adaptive_scheduler/scheduler.py

```diff
@@ -209,7 +209,7 @@ def save_schedule(self, schedule, estimated_scheduler_end, semester_details, is_
             now = datetime.utcnow()
             day_timestamp = now.strftime('%Y-%m-%d')
             file_timestamp = now.strftime('%Y%m%d%H%M%S')
-            filename = '{}_schedule_{}.json'.format(schedule_type, file_timestamp)
+            filename = '{}_{}_schedule_{}.json'.format(schedule_type, '_'.join(self.sched_params.telescope_classes), file_timestamp)
```

Comment:
If `telescope_classes` is empty, there is going to be a double underscore. I would handle that case, maybe by making the telescope classes in the filename be "all" or making sure there isn't a double underscore.

Reply:
good call